src/highlightjs-adapter.ts

```diff
@@ -1,36 +1,36 @@
 const {Opal} = require('asciidoctor-opal-runtime')
 
-module.exports.register = (asciidoctor) => {
-  const HighlightjsSyntaxHighlighter = asciidoctor.SyntaxHighlighter.for('highlight.js')
-  const customHighlightJsAdapter = Opal.klass(Opal.nil, HighlightjsSyntaxHighlighter, 'CustomHighlightJsAdapter')
+module.exports.register = (highlightjsBuiltInSyntaxHighlighter, context: vscode.ExtensionContext, webviewPanel: vscode.WebviewPanel) => {
+  const customHighlightJsAdapter = Opal.klass(Opal.nil, highlightjsBuiltInSyntaxHighlighter, 'CustomHighlightJsAdapter')
   customHighlightJsAdapter.$register_for('highlight.js', 'highlightjs')
+
   let $docinfo
-  const baseUrl = 'https://cdnjs.cloudflare.com/ajax/libs/highlight.js/9.18.3'
   Opal.def(customHighlightJsAdapter, '$docinfo', $docinfo = function $$docinfo(location, doc, opts) {
-    const self = this;
+    const self = this
     if (location === 'head') {
       const theme = doc.$attr('highlightjs-theme', 'github')
-      const integrityValue = sri[`styles/${theme}.min.css`]
-      return `<link rel="stylesheet" href="${baseUrl}/styles/${theme}.min.css" ${integrityValue ? `integrity="${integrityValue}" ` : ''}crossorigin="anonymous" referrerpolicy="no-referrer">`
+      const themeStyleSheetResource = vscode.Uri.file(path.join(context.extensionPath, 'media', 'highlightjs', 'styles', `${theme}.min.css`))
+      return `<link rel="stylesheet" href="${webviewPanel.webview.asWebviewUri(themeStyleSheetResource)}">`
     }
     // footer
     let languageScripts = ''
     if (doc['$attr?']('highlightjs-languages')) {
       languageScripts = doc.$attr('highlightjs-languages').split(',').map((lang) => {
-        const key = `languages/${lang.trim()}.min.js`
-        const integrityValue = sri[key]
-        return `<script src="${baseUrl}/${key}" ${integrityValue ? `integrity="${integrityValue}" ` : ''}crossorigin="anonymous" referrerpolicy="no-referrer"></script>`
+        const languageScriptResource = vscode.Uri.file(path.join(context.extensionPath, 'media', 'highlightjs', 'languages', `${lang.trim()}.min.js`))
+        return `<script src="${webviewPanel.webview.asWebviewUri(languageScriptResource)}"></script>`
       }).join('\n')
     }
-    return `<script src="${baseUrl}/highlight.min.js" integrity="${sri['highlight.min.js']}" crossorigin="anonymous" referrerpolicy="no-referrer"></script>
+    const highlightjsScriptResource = vscode.Uri.file(path.join(context.extensionPath, 'media', 'highlightjs', 'highlight.min.js'))
+    return `<script src="${webviewPanel.webview.asWebviewUri(highlightjsScriptResource)}"></script>
 ${languageScripts}
 <script>
 if (!hljs.initHighlighting.called) {
   hljs.initHighlighting.called = true
-  ;[].slice.call(document.querySelectorAll("pre.highlight > code")).forEach(function (el) { hljs.highlightBlock(el) })
+  ;[].slice.call(document.querySelectorAll("pre.highlight > code")).forEach(function (el) { hljs.highlightElement(el) })
 }
 </script>`
```

Comment:
Will this ever be called if security is not set to disabled?

Reply:
What do you mean by "if security is not set to disabled?"

The hash `'sha256-ZrDBcrmObbqhVV/Mag2fT/y08UJGejdW7UWyEsi4DXw='` is declared in all CSP (even when the security level is "strict").

Reply:
I was having a moment, all good!